File: WebSite/WebSite.Common/Dto/ResultDto.cs
namespace WebSite.Common.Dto
{
    public class ResultDto
    {
        public bool IsSuccess { get; }

        public string Message { get; }

        public ResultDto(bool isSuccess, string message)
        {
            IsSuccess = isSuccess;
            Message = message;
        }
    }

    public class ResultDto<T>
    {
        public bool IsSuccess { get; }

        public string Message { get; }

        public T Data { get; }

        public ResultDto(bool isSuccess, string message, T data)
        {
            IsSuccess = isSuccess;
            Message = message;
            Data = data;
        }
    }
}

File: rules-evaluator/src/main/java/edu/sharif/ce/rules_evaluator/httpClient/ResultDto.java
package edu.sharif.ce.rules_evaluator.httpClient;

public class ResultDto {

    public boolean isSuccess;

    public String message;

    public ResultGetRuleListDto data;

    public boolean isSuccess() {
        return isSuccess;
    }

    public void setSuccess(boolean success) {
        isSuccess = success;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public ResultGetRuleListDto getData() {
        return data;
    }

    public void setData(ResultGetRuleListDto data) {
        this.data = data;
    }
}
